libraries/ms-helper/inc/thermistor_converter.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+// Thermistor Reading interface - converts an ADC voltage reading to temperature
+// Requires GPIO and Interrupts to be initialized.
+#include "adc.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "soft_timer.h"
+
+#define RESISTANCE_LEN 71
+
+typedef struct {
+  uint32_t sibling_resistance;  // milliohms
+  uint32_t source_voltage;      // millivolts
+  void *context;
+
+  ADCChannel adc_channel;
+} ThermistorStorage;
+
+typedef struct {
+  uint32_t sibling_resistance;  // milliohms
+  uint32_t source_voltage;      // millivolts
+  void *context;
+
+  GPIOSettings *gpio_settings;
+  GPIOAddress *gpio_addr;
+  ADCMode adc_mode;
+  ADCChannel adc_channel;
+} ThermistorSettings;
```

Comment:
If your module is called `thermistor_converter`, these should be prefixed with `ThermistorConverter`.

I would suggest changing your module name to something a little easier to say. There's no real reason it couldn't just be `thermistor`.